ethcore/src/client/client.rs

```diff
@@ -122,11 +122,13 @@ impl SleepState {
 /// Call `import_block()` to import a block asynchronously; `flush_queue()` flushes the queue.
 pub struct Client {
 	mode: Mode,
-	chain: Arc<BlockChain>,
-	tracedb: Arc<TraceDB<BlockChain>>,
+	chain: RwLock<Arc<BlockChain>>,
```

Comment:
But you can always just clone the `Arc` instead of accessing the shared pointer. Maybe it's worthing wrapping `Arc<BlockChain>` into a structure with `Deref` but without `Clone` to prevent such situations?